### PR TITLE
statistics: reduce the copy when to merge global status

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #47219

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/table.go [hunk 1]
```diff
@@ -377,18 +377,26 @@ func (t *Table) ColumnByName(colName string) *Column {
 }
 
 // GetStatsInfo returns their statistics according to the ID of the column or index, including histogram, CMSketch, TopN and FMSketch.
-func (t *Table) GetStatsInfo(id int64, isIndex bool) (*Histogram, *CMSketch, *TopN, *FMSketch, bool) {
+func (t *Table) GetStatsInfo(id int64, isIndex bool, needCopy bool) (*Histogram, *CMSketch, *TopN, *FMSketch, bool) {
```
Comment (winoros) [inline statistics/table.go:380]: Add comments for the newly added parameter.

Comment (hawkingrei) [reply statistics/table.go:380]: Done

**Review by winoros — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Comment by L-maple** [comment]
@hawkingrei Hi, I try to understand the global stats recently. 
I see this PR, but do not understand why cache is useless. In my  opinion, cache can reduce load stats from storage. 
For example, a partitioned table(10 partitions) has 4 columns and 4 indexes. 
Before, 10 loadStats. After, 5 * 10 loadStats.

**Pull request merged**